builtin/credential/aws-ec2/path_login.go

```diff
@@ -265,60 +320,97 @@ func (b *backend) parseIdentityDocument(s logical.Storage, pkcs7B64 string) (*id
 // option is enabled on the registered role.
 func (b *backend) pathLoginUpdate(
 	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	identityDocB64 := data.Get("identity").(string)
```

Comment:
Given that in the identity verification function you convert what comes in to `[]byte` (after first converting to a `string`), why not simply use the base64-decoded value directly as []byte? For the signature at least this should be raw bytes anyways as it's already base64-encoded from the metadata endpoint AFAIK. (Or are you expecting the caller to base64-encode the base64-encoded value, which I'm not sure does the right thing on decode?) For the identity doc my guess is that you're concerned about whitespace but if they pipe the output from a curl call directly into base64 it ought to be correct.